src/views/datalist/data-list-directive.js

```diff
@@ -227,6 +279,80 @@ angular.module('patternfly.views').directive('pfDataList', function (pfUtils) {
             'Illegal use of pfDataList directive! ' +
             'Cannot allow both select box and click selection in the same data list.');
           }
+
+          if (!$scope.listId) {
+            $scope.listId = 'pf-data-list';
```

Comment:
See comment about constructing HTML in javascript below.  If we were to use angular-bootstrap's [dropdown directive](https://angular-ui.github.io/bootstrap/#/dropdown) then I think the need for `$scope.listId` is removed.